subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -78,6 +92,16 @@ public BuildScriptBuilder fileComment(String comment) {
         return Splitter.on("\n").splitToList(comment.trim());
     }
 
+    private static URL urlFromString(String urlAsString) {
+        final URL urlAsURL;
+        try {
+            urlAsURL = new URL(urlAsString);
+        } catch (MalformedURLException e) {
+            throw new GradleException(String.format("Invalid URL: '%s'.", urlAsString), e);
```

Comment:
I would use an explicitly unchecked exception here (GradleException is unchecked too, but this is what we do in other places for this same exception):

```suggestion
                        throw UncheckedException.throwAsUncheckedException(e);
```